include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -435,71 +435,28 @@ struct __sycl_usm_alloc
     }
 };
 
-//-----------------------------------------------------------------------
-// type traits for objects granting access to some value objects
-//-----------------------------------------------------------------------
-
-template <typename _ContainerOrIterator>
-struct __memobj_traits
-{
-    using value_type = typename _ContainerOrIterator::value_type;
-};
-
-template <typename _T>
-struct __memobj_traits<_T*>
-{
-    using value_type = _T;
-};
-
-} // namespace __internal
-
-template <typename _ExecutionPolicy, typename _T>
-using __buffer = __internal::__buffer_impl<::std::decay_t<_ExecutionPolicy>, _T>;
-
-template <typename T>
-struct __repacked_tuple
-{
-    using type = T;
-};
-
-template <typename... Args>
-struct __repacked_tuple<::std::tuple<Args...>>
-{
-    using type = oneapi::dpl::__internal::tuple<Args...>;
-};
-
-template <typename T>
-using __repacked_tuple_t = typename __repacked_tuple<T>::type;
-
-template <typename _ContainerOrIterable>
-using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
-
 template <typename _T>
 struct __accessor
 {
   private:
     using __accessor_t = sycl::accessor<_T, 1, sycl::access::mode::read_write, __dpl_sycl::__target_device,
                                         sycl::access::placeholder::false_t>;
-    __accessor_t __acc;
-    _T* __ptr = nullptr;
+
     bool __usm = false;
+    _T* __ptr = nullptr;
+    __accessor_t __acc;
```

Comment:
As in previous code, initialization of `__usm` member is the first, so this member should be at first position inside class layout as far as we implement initialization of another members inside initialization list of class constructor.

Reply:
To prevent warnings, you can change the order in the init list - this is OK because `__usm` is not used to initialize other members.

Reply:
Of course. it's possible.
But from my point of view, `__usm` is the `boss` inside this class, so better to initialize it at first and have it at first place in class fields.

Reply:
Besides warnings, it sounds like there is no solid technical arguments justifying your preference to put `__usm` first in the class layout. On the other hand, the ordering before the patch follows the rule of ordering class members from most aligned to least aligned, as this usually minimizes paddings in the class layout, and so possibly reduces its size.

For these very classes, there might be no difference as of now. But since no strong justification to change the order is provided, I prefer to fix the warnings by changing the member initialization order instead.